src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -142,7 +177,30 @@ public void Write<T>([AllowNull] T value)
         /// <paramref name="npgsqlDbType"/> must be specified as <see cref="NpgsqlDbType.Jsonb"/>.
         /// </param>
         /// <typeparam name="T">The .NET type of the column to be written.</typeparam>
-        public void Write<T>([AllowNull] T value, NpgsqlDbType npgsqlDbType)
+        public void Write<T>([AllowNull] T value, NpgsqlDbType npgsqlDbType) =>
+            Write(value, npgsqlDbType, false).GetAwaiter().GetResult();
+
+        /// <summary>
+        /// Writes a single column in the current row as type <paramref name="npgsqlDbType"/>.
+        /// </summary>
+        /// <param name="value">The value to be written</param>
+        /// <param name="npgsqlDbType">
+        /// In some cases <typeparamref name="T"/> isn't enough to infer the data type to be written to
+        /// the database. This parameter and be used to unambiguously specify the type. An example is
+        /// the JSONB type, for which <typeparamref name="T"/> will be a simple string but for which
+        /// <paramref name="npgsqlDbType"/> must be specified as <see cref="NpgsqlDbType.Jsonb"/>.
+        /// </param>
+        /// <param name="cancellationToken"></param>
+        /// <typeparam name="T">The .NET type of the column to be written.</typeparam>
+        public Task WriteAsync<T>([AllowNull] T value, NpgsqlDbType npgsqlDbType, CancellationToken cancellationToken = default)
+        {
+            if (cancellationToken.IsCancellationRequested)
+                return Task.FromCanceled(cancellationToken);
+            using (NoSynchronizationContextScope.Enter())
+                return Write(value, npgsqlDbType, true);
+        }
+
+        async Task Write<T>([AllowNull] T value, NpgsqlDbType npgsqlDbType, bool async)
```

Comment:
This method doesn't need to be async - it can just return the task from Write on the last line.